src/bindings/python/flux/job.py

```diff
@@ -98,6 +98,32 @@ def wait(flux_handle, jobid=lib.FLUX_JOBID_ANY):
     return status
 
 
+# Due to subtleties in the python bindings and this call, this binding
+# is more of a reimplementation of flux_job_list().  Subtle issues
+# are:
+#
+# - Python Future class does not have get()
+# - Desired return value is json array, not a single value
+#
+# pylint: disable=dangerous-default-value
+def job_list(flux_handle, max_entries=0, attrs=[], userid=os.geteuid(), flags=0):
+    payload = {
+        "max_entries": max_entries,
+        "attrs": attrs,
+        "userid": userid,
+        "flags": flags,
+    }
+    return flux_handle.rpc("job-info.list", payload)
+
+
+def job_list_get(future_handle):
+    try:
+        ret = future_handle.get()
+    except EnvironmentError as err:
+        raise err
```

Comment:
I may be wrong, but I don't think this try/except block buys you anything.  It catches the exception and then re-raises it immediately.  I'd just drop the try/except.